scipy/stats/_multivariate.py

```diff
@@ -4563,12 +4564,32 @@ def _entropy(self, dim, df=1, shape=1):
             return multivariate_normal(None, cov=shape).entropy()
 
         shape_info = _PSD(shape)
-        halfsum = 0.5 * (dim + df)
-        half_df = 0.5 * df
-        return (-gammaln(halfsum) + gammaln(half_df)
+        shape_term = 0.5 * shape_info.log_pdet
+
+        def regular(dim, df):
+            halfsum = 0.5 * (dim + df)
+            half_df = 0.5 * df
+            return (
+                -gammaln(halfsum) + gammaln(half_df)
                 + 0.5 * dim * np.log(df * np.pi) + halfsum
                 * (psi(halfsum) - psi(half_df))
-                + 0.5 * shape_info.log_pdet)
+                + shape_term
+            )
+
+        def asymptotic(dim, df):
+            # Formula from Wolfram Alpha:
+            # "asymptotic expansion -gammaln((m+d)/2) + gammaln(d/2) + (m*log(d*pi))/2
+            #  + ((m+d)/2) * (digamma((m+d)/2) - digamma(d/2))"
```

Comment:
To double-check, I copy-pasted the regular formula above and made appropriate replacements. I got something equivalent.
```
asymptotic expansion -gammaln((m+d)/2) + gammaln(d/2) + m/2*log(d*pi) + (m+d)/2 * (digamma((m+d)/2) - digamma(d/2))
```
Wolfram Alpha gives
![image](https://github.com/scipy/scipy/assets/6570539/964b452e-8da0-4c42-bad9-cb14e4799df8)

before some higher order terms, and that seems to be what you've coded below. I'm not concerned about `dim` to high powers because without sparse arrays it's impossible for `dim` to be high enough to cause overflows.